git-authors: add page (#5951)
tldr-pages/tldr: pages/common/git-authors.md

# git authors

> > Display log of authors of a Git repository.

Reviewer: ```suggestion
> Display list of commiters of a Git repository, sorted by author's commit count.
```

> > A part of git-extras.

Reviewer: ```suggestion
> Part of `git-extras`.
```

> > Display log or list of committers of a Git repository. Everything except `--list` will create a new file called `AUTHORS`.

Reviewer 2: Perhaps this will read better?
```suggestion
> Display a log or list of committers of a Git repository. Everything except `--list` will create a new file called `AUTHORS`.
```

> > Display log or list of committers of a Git repository. If you aren't using the list command, a new file called AUTHORS will be created. Sorted by amount of commits.

Reviewer 3: ```suggestion
> Display log or list of committers of a Git repository.
```
This doesn't really seem important.

Author: It does to me, since otherwise someone will be confused when a new random file appears in their directory

Reviewer 3: How about that? (We should avoid `you`, `yours`, etc.)
```suggestion
> Display log or list of committers of a Git repository. Everything except `--list` will create a new file called `AUTHORS`.
```

> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-authors>.

> - Display log of authors/committers of a Git repository:

Reviewer: I tried it; this command does a bit more:
```suggestion
- Create a new (or update the existing) `AUTHORS` file in the current directory, populate it from the current Git repository, and open it in the default editor:
```

Author: I really did wonder about that file, I thought it was the command editing the AUTHORS file, turns out it didn't exist!

Thanks for that

`git authors`

> - Print full list of authors/committers of a Git repository:

Reviewer: ```suggestion
- Display the full authors list for the current Git repository:
```

> - Display a log of committers in the default editor, reading from the `AUTHORS` file:

Reviewer 4: This command create or update the `AUTHORS` file:
```suggestion
- Create or replace the `AUTHORS` file with the list of committers and open it in the default editor:
```

> - Print a full list of committers:

Reviewer 4: ```suggestion
- Display a full list of committers:
```

Reviewer 3: > Also the the usage of display and print is used since the display to me indicates that it isn't a firm output, rather something that can be scrolled through, while the print is used to indicate that it just dumps the entire lot, as if it was concatenated using cat

I kind of have to agree with @CleanMachine1 here

Author: It is print since it dumps it, display to me indicates more of a elegant way of displaying.
the display indicates that it is opened in a text editor

Author: > > Also the the usage of display and print is used since the display to me indicates that it isn't a firm output, rather something that can be scrolled through, while the print is used to indicate that it just dumps the entire lot, as if it was concatenated using cat
> 
> I kind of have to agree with @CleanMachine1 here

couldn't find my reply! Was looking for it

> - Display log of committers of the current Git repository, reading from the AUTHORS file, opened in your default editor:

Reviewer 3: ```suggestion
- Display a log of committers in the default editor, reading from the `AUTHORS` file:
```

Author: The editor is required since the list version doesn't

Reviewer 3: I haven't removed it 🤔

Author: so you haven't

Author: ok then commited

`git authors --list`

> - Print full list of authors/committers of a Git repository excluding emails:

Reviewer: ```suggestion
- Display the full authors list for the current Git repository, excluding emails:
```

> - Print full list of committers of the current Git repository:

Reviewer 3: ```suggestion
- Print a full list of committers:
```
I don't think that's neccecary since it's the only way it makes sense.

Author: Ok yes

> - Display a log of committers, exluding emails, in the default editor, reading from the `AUTHORS` file:

Reviewer 4: Why log?
```suggestion
- Display a full list of committers, excluding emails, in the default editor, reading from the `AUTHORS` file:
```
(a typo here)

Reviewer 3: Good question. It is indeed only a plain list.

Author: < Why log? 

because the list is used in the list variant of the command.

Reviewer 4: > because the list is used in the list variant of the command.

I didn't get this.

Reviewer 3: But it's also a plain list when you run `git authors`.

Author: For me it opens in VIM, since it opens in the text editor

Author: --list makes a dumped list

Reviewer 4: I think a `log` is a record of events, in this case were are not talking about events of any kind.

Reviewer 3: But what does that have to do with a log? To me a log is something, where changes or events are saved with a timestamp, like `git log {{file}}`.

Author: To me a log, is a list of events as well, however it isn't required to be timestamped

Reviewer 3: `git authors` only outputs a list of contributors, which I wouldn't call a log, but just a list.

Author: What happens exactly, the same as --list with it?

Reviewer 4: `--list` don't create the `AUTHORS` file.

> - Display a log of committers, excluding emails, in the default editor, reading from the `AUTHORS` file:

Reviewer 4: This is not a log of events, is just a list.
```suggestion
- List committers, excluding emails, in the default editor, reading from the `AUTHORS` file:
```

> - List committers, excluding emails, in the default editor, reading from the `AUTHORS` file:

Reviewer 2: Perhaps this will read better?
```suggestion
- Launch the default editor to display a list of committers (excluding emails) from the `AUTHORS` file:
```

Author: I disagree with this one, it sounds wrong

Reviewer 4: Maybe sounds wrongs, but @sbrl is right, is not a list in terminal, Launch an app with something. :thinking: @bl-ue ?

Reviewer: To me they both imply basically the same meaning.

Reviewer 3: I think too. When you list something in the default editor, I think that implies that it launches it first. I prefer the first one.

Reviewer 2: Hrm, I see the concerns here. However this PR has been prematurely merged, which limits the usefulness of me suggesting anything else here. Regardless, what about this?
```suggestion
- Display a list of committers (excluding emails) from the `AUTHORS` file in the default text editor:
```

> - Display full log of committers of the current Git repository, excluding emails, reading from the AUTHORS file, opened in your default editor:

Reviewer 3: ```suggestion
- Display a log of committers, exluding emails, in the default editor, reading from the `AUTHORS` file:
```

Author: why does this have the default editor bit but the other suggestion doesn't?
Consistency isn't present here

# git authors

> > Display log or list of committers of a Git repository. Sorted by amount of commits.
> > Part of `git-extras`.
> > Creates an AUTHORS file in the Git repository, this will be used for the command.

Reviewer: Unfortunately, we can't have more thant 3 total description lines (including the link) :/
```suggestion
> Display log or list of committers of a Git repository. Sorted by amount of commits.
> Part of `git-extras`.
```
The `AUTHORS` line isn't even true for all possible invocations of the command (running it with `--list` doesn't operate on `AUTHORS`, I don't think)

See https://github.com/tldr-pages/tldr/pull/5807#discussion_r617511072 for details.

Author: You're right, the list doesn't use a file

Author: take a look at the commit after this
